### PR TITLE
Added padding option for ACSF

### PR DESCRIPTION
**Pull request by andersx**
Adding padding option for ACSF in preparation for gradient kernel functions.

### CODE REVIEW

**Review by charnley — DISMISSED** [review]

**Review by larsbratholm — CHANGES_REQUESTED** [review]
Looks fine. Can you improve the doc explanation of what the `pad` parameter does? I found it a bit confusing that it doesn't say that `pad` should be the largest number of atoms in your training set. Maybe `pad` could be called `max_natoms` or something.
Also there should be a check if `natoms > pad`. Finally, it would make more sense for me to do `if gradients: ... else` and `if pad is None: ... else` rather than the other way around, but doesn't matter much.
